core/downloadfromdcthread.cpp

```diff
@@ -287,12 +287,8 @@ QStringList DCDeviceData::getProductListFromVendor(const QString &vendor)
 
 int DCDeviceData::getMatchingAddress(const QString &vendor, const QString &product)
 {
-	for (int i = 0; i < connectionListModel.rowCount(); i++) {
-		QString address = connectionListModel.address(i);
-		if (address.contains(product))
-			return i;
-	}
-	return -1;
+	Q_UNUSED(vendor)
+	return connectionListModel.indexOf(product);
```

Comment:
I think you replaced a substring by a full-match search. Not sure what is correct here.

Reply:
look at our implementation of `indexOf()`:
```
int ConnectionListModel::indexOf(QString address)
{
	const QRegExp re(".*" + address + ".*", Qt::CaseInsensitive);
	return m_addresses.indexOf(re);
}
```